Reinstate the wpt lint checking.

Ms2ger opened the pull request:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/servo/servo/13999)
<!-- Reviewable:end -->

Ms2ger committed:
Reinstate the wpt lint checking.

highfive commented:
Heads up! This PR modifies the following files:
 * @wafflespeanut: python/tidy/servo_tidy/tidy.py

Ms2ger committed:
Remove trailing whitespace from a test.

Cherry-pick of <https://github.com/w3c/web-platform-tests/commit/f23222137d3fbc14fae1518e64ab3aa9bbf32a7c>.

Ms2ger committed:
Add a lint whitelist for our own tests.

jdm commented:
`./mach test-tidy --all` is unhappy.

Ms2ger committed:
Add missing path to the path length error.

Partial cherry-pick of <https://github.com/w3c/wpt-tools/pull/136>.

Ms2ger commented:
Fixed.

Ms2ger commented on python/tidy/servo_tidy/tidy.py:
```diff
@@ -782,8 +782,8 @@ def collect_errors_for_files(files_to_check, checking_functions, line_checking_f
-def get_wpt_files(only_changed_files, progress):
-    wpt_dir = os.path.join(".", "tests", "wpt" + os.sep)
+def get_wpt_files(suite, only_changed_files, progress):
+    wpt_dir = os.path.join(".", "tests", "wpt", suite, "")
```
I'd prefer not doing that in this PR.

Wafflespeanut commented:
@bors-servo r+

bors-servo commented:
:pushpin: Commit 2ff0f33 has been approved by `Wafflespeanut`

<!-- @bors-servo r=Wafflespeanut 2ff0f33b2b1caaef6d2b504ccc5f04f401418f86 -->

bors-servo commented:
:hourglass: Testing commit 2ff0f33 with merge cb8ac3f...

bors-servo commented:
:sunny: Test successful - [arm32](http://build.servo.org/builders/arm32/builds/3806), [arm64](http://build.servo.org/builders/arm64/builds/3814), [linux-dev](http://build.servo.org/builders/linux-dev/builds/3892), [linux-rel-css](http://build.servo.org/builders/linux-rel-css/builds/532), [linux-rel-wpt](http://build.servo.org/builders/linux-rel-wpt/builds/531), [mac-dev-unit](http://build.servo.org/builders/mac-dev-unit/builds/3883), [mac-rel-css](http://build.servo.org/builders/mac-rel-css/builds/3875), [mac-rel-wpt1](http://build.servo.org/builders/mac-rel-wpt1/builds/564), [mac-rel-wpt2](http://build.servo.org/builders/mac-rel-wpt2/builds/570), [windows-dev](http://build.servo.org/builders/windows-dev/builds/1796)
